### PR TITLE
Qsw 683 - estimate wall clock

### PR DESCRIPTION
**Pull request by claudio525**
Python3 compatibility changes.

### CODE REVIEW

File: qcore/geo.py [hunk 1]
```diff
@@ -2,24 +2,27 @@
 Various tools which may be needed in various processes.
 """
 
-from math import sin, asin, cos, atan, atan2, degrees, radians, sqrt, pi
 import os
+import numpy as np
```
Comment (viktor76525) [inline qcore/geo.py]: import builtin1 (os, sys, math, subprocess)
import builtin2

import external1 (numpy, h5py etc..)
import external2

import local1 (qcore or other local files)
import local2

i try to go with the above format, alphabetical within blocks, not sure why they have been rearranged

File: qcore/geo.py [hunk 2]
```diff
@@ -2,24 +2,27 @@
...
 class InputError(Exception):
     pass
 
-def ll2gp_multi(coords, mlon, mlat, rot, nx, ny, hh, \
-        dx = 1, dy = 1, decimated = False, verbose = False, \
-        keep_outside = False):
+
+def ll2gp_multi(coords, mlon, mlat, rot, nx, ny, hh,
+                dx=1, dy=1, decimated=False, verbose=False,
+                keep_outside=False):
```
Comment (viktor76525) [inline qcore/geo.py]: just run "black code.py" (https://github.com/ambv/black) and it will format without bothering about it

File: qcore/geo.py [hunk 3]
```diff
@@ -38,16 +41,18 @@ def ll2gp_multi(coords, mlon, mlat, rot, nx, ny, hh, \
...
-    p_conv = Popen(cmd,
-            stdin = PIPE, stdout = PIPE)
-    stdout = p_conv.communicate( \
-            '\n'.join(['%s %s' % tuple(c) for c in coords]))[0]
-    xy = [map(float, line.split()) for line in stdout.rstrip().split('\n')]
+
+    # Has to be a byte string
+    p_conv = Popen(cmd, stdin=PIPE, stdout=PIPE)
+    stdout = p_conv.communicate(
+        '\n'.join(['%s %s' % tuple(c) for c in coords]).encode())[0]
```
Comment (viktor76525) [inline qcore/geo.py]: this will produce a byte array. need to run .decode() after too.

File: qcore/geo.py [hunk 4]
```diff
@@ -38,16 +41,18 @@ def ll2gp_multi(coords, mlon, mlat, rot, nx, ny, hh, \
...
+    xy = [list(map(float, line.split()))
+          for line in stdout.rstrip().split(b'\n')]
```
Comment (viktor76525) [inline qcore/geo.py]: i guess considering the final output is actually a float, converted then it's ok to not run decode as long as this part works fine.

File: qcore/srf.py [hunk 5]
```diff
@@ -9,21 +9,21 @@
...
-from math import ceil, cos, floor, radians, sqrt
 import os
-from subprocess import Popen, PIPE
-
 import numpy as np
 
-from qcore import geo
-from qcore.config import qconfig
+from math import ceil, cos, floor, radians, sqrt, sin, degrees, atan
+from subprocess import Popen, PIPE
```
Comment (viktor76525) [inline qcore/srf.py]: again, the built in python modules (os, math, subprocess) have been put in separate blocks when they belong together.

File: qcore/geo.py (hunk 1 above)
Comment (sungeunbae) [reply qcore/geo.py]: Sounds good, but should we enforce this rule? I think we need to have a set of internal rules that everyone can agree and follow

File: qcore/geo.py [hunk 6]
```diff
@@ -115,22 +122,26 @@ def gp2ll_multi(coords, mlat, mlon, rot, nx, ny, hh):
...
-            stdin = PIPE, stdout = PIPE)
-    stdout = p_conv.communicate( \
-            '\n'.join(['%s %s' % tuple(c) for c in coords]))[0]
+    p_conv = Popen([xy2ll_bin, 'mlat=%s' % (mlat), 'mlon=%s' % (mlon),
+                    'geoproj=1', 'center_origin=1', 'h=%s' % (hh),
+                    'xazim=%s' % (xazim), 'xlen=%s' % (xlen),
+                    'ylen=%s' % (ylen)],
+                   stdin=PIPE, stdout=PIPE)
+    stdout = p_conv.communicate(
+            '\n'.join(['%s %s' % tuple(c) for c in coords]).encode())[0]
```
Comment (viktor76525) [inline qcore/geo.py]: missed one (to keep consistent)

**Pull request merged**